### PR TITLE
2D Hidden visualization

### PR DESCRIPTION
**Pull request by Lucaskabela**
Add a python file for 2d visualization of latent space.

To run, simply
`python viz.py --config ./config/file`

This script will visualize the 5 furthest (measured by Euclidean distance between paired speech and text) hidden vectors after PCA to 2 dimensions, normalization, and some small noise to add seperation

### CODE REVIEW

File: src/viz.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        else:
+            t_out = text_latent
+            s_out = speech_latent        
+        text_latent = t_out[:, -1, :]
+        speech_latent = s_out[:, -1, :]
+        combined = torch.cat([text_latent, speech_latent], dim=0).cpu().numpy()
+        break
+    print(combined.shape)
+    latents = StandardScaler().fit_transform(combined)
+    pca = PCA(n_components=2)
```
Comment (billyang98) [inline src/viz.py:84]: can similarly try t-sne here https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html

>>> import numpy as np
>>> from sklearn.manifold import TSNE
>>> X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
>>> X_embedded = TSNE(n_components=2).fit_transform(X)
>>> X_embedded.shape
(4, 2)

File: src/viz.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            text_latent, _ = model.text_m.encode(text, text_len)
+            speech_latent, _ = model.speech_m.encode(mel, mel_len)
+
+        if args.model_type == 'rnn':
+            _, t_out = text_latent
+            _, s_out = speech_latent
+        else:
+            t_out = text_latent
+            s_out = speech_latent        
+        text_latent = t_out[:, -1, :]
```
Comment (billyang98) [inline src/viz.py:78]: try the torch.mean(t_out, dim=1)

Comment (billyang98) [reply src/viz.py:78]: oh also since these are from the same batch, the technically have the same size, could just run t-sne/pca on 
t_out directly.
